File: lib/models/amap_subpoi.dart
import 'package:amap_flutter_search/models/amap_geo_point.dart';
import 'package:json_annotation/json_annotation.dart';
part 'amap_subpoi.g.dart';

@JsonSerializable()
class AMapSubPOI {
  ///POI全局唯一ID
  String? uid;

  ///名称
  String? name;

  ///名称简写
  String? sname;

  ///经纬度
  AMapGeoPoint? location;

  ///地址
  String? address;

  ///距中心点距离
  int? distance;

  ///子POI类型
  String? subtype;

  AMapSubPOI();

  factory AMapSubPOI.fromJson(Map<String, dynamic> srcJson) =>
      _$AMapSubPOIFromJson(srcJson);

  Map<String, dynamic> toJson() => _$AMapSubPOIToJson(this);
}

File: android/src/main/java/com/amap/amap_flutter_search/AMapSubPOI.java
package com.amap.amap_flutter_search;

public class AMapSubPOI {
    ///POI全局唯一ID
    private String uid;

    ///名称
    private String name;

    ///名称简写
    private String sname;

    ///经纬度
    private AMapGeoPoint location;

    ///地址
    private String address;

    ///距中心点距离
    private int distance;

    ///子POI类型
    private String subtype;

    public String getUid() {
        return uid;
    }

    public void setUid(String uid) {
        this.uid = uid;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getSname() {
        return sname;
    }

    public void setSname(String sname) {
        this.sname = sname;
    }

    public AMapGeoPoint getLocation() {
        return location;
    }

    public void setLocation(AMapGeoPoint location) {
        this.location = location;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public int getDistance() {
        return distance;
    }

    public void setDistance(int distance) {
        this.distance = distance;
    }

    public String getSubtype() {
        return subtype;
    }

    public void setSubtype(String subtype) {
        this.subtype = subtype;
    }
}
